activation/activation.go

```diff
@@ -383,60 +324,57 @@ func (b *Builder) loadChallenge() error {
 	return nil
 }
 
-func (b *Builder) discardChallenge() error {
-	return b.store.Put(b.getNipstKey(), []byte{})
-}
-
-// PublishActivationTx attempts to publish an atx for the given epoch, it returns an error if an atx cannot be created
-// publish atx may not produce an atx each time it is called, that is expected behaviour as well.
-func (b *Builder) PublishActivationTx(epoch types.EpochId) error {
-	if b.nipst != nil {
-		b.log.With().Info("re-entering atx creation in epoch %v", log.EpochId(uint64(epoch)))
+// PublishActivationTx attempts to publish an atx, it returns an error if an atx cannot be created.
+func (b *Builder) PublishActivationTx() error {
+	b.refreshChallenge()
```

Comment:
why is it called refreshChallenge?

Reply:
Feel free to suggest a better name.

What this method does is ensure that the challenge is "fresh" (the target epoch hasn't passed) and discards the challenge (and logs it) otherwise.

I could call it `discardChallengeIfStale`, wdyt?

Reply:
I've renamed it to `discardChallengeIfStale` - I think it's really clearer this way